### DISCUSSION TITLE
F-strings doesn't work in templating

### DISCUSSION THREAD

**Discussion by V0lantis** [apache/airflow#19537] [Q&A] [answered]
Hello,

I am trying to use airflow in some projects. I was trying to change the following pattern : 

```python
"{{ '{2}'.format(execution_date.month) }}"
```

by the following pattern : 

```python
"{{ f"{execution_date.month:2}" }}"
```

But it doesn't work. Some idea why it doesn't work?

Full code : 

```python
get_data = BashOperator(
    task_id="get_data",
    bash_command=(
        "curl -o /tmp/wikipageviews.gz "
        "echo https://dumps.wikimedia.org/other/pageviews/"
        "{{ execution_date.year }}/"
        "{{ execution_date.year }}-"
        "{{ '{:02}'.format(execution_date.month) }}/
       ...
```

**Comment by josh-fell** [comment, marked as answer]
For this case you could use the `strftime()` function on `execution_date`. Something like this should work:
```python
get_data = BashOperator(
        task_id="get_data",
        bash_command=(
            "curl -o /tmp/wikipageviews.gz "
            "echo https://dumps.wikimedia.org/other/pageviews/"
            "{{ execution_date.year }}/"
            "{{ execution_date.year }}-"
            "{{ execution_date.strftime('%m') }}/"
        ),
    )
```

**Reply by V0lantis** [reply]
My question doesn't resume itself to this specific case. I would like to know if it is possible to use f-string inside Jinja template in airflow, and if yes, how?

**Reply by potiuk** [reply]
> My question doesn't resume itself to this specific case. I would like to know if it is possible to use f-string inside Jinja template in airflow, 

No it does not work - try it yourself on plain JINJA templating and see it yourself.

> and if yes, how?

Josh answered it nicely,

**Reply by V0lantis** [reply]
Thank you both for your answer. 

> No it does not work - try it yourself on plain JINJA templating and see it yourself.

Yes, I did and I got a strange error message from airflow. I tried to search Airflow and Jijna doc about f-string and didn't find anything. Also tried to look for some on stackoverflow without success.

> Josh answered it nicely,

Absolutely, and I apologize sincerely if my question was rude. It was not meant to be.
